src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -199,44 +286,92 @@ public void PacketReset(SNIHandle handle, bool write, SNIPacket packet)
+                    string serverNameWithOutPortOrInstanceName = portOrInstanceNameIndex > 0 ? fullServerName.Substring(0, portOrInstanceNameIndex) : fullServerName;
+                    IPAddress address = null;
 
-                default:
-                    if (parallel)
+                    // when no protocol is specified, and fullServerName is IPv6
+                    if (IPAddress.TryParse(serverNameWithOutPortOrInstanceName, out address) && address.AddressFamily == AddressFamily.InterNetworkV6)
                     {
-                        SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
+                        // default to using tcp if no protocol is provided
+                        sniHandle = CreateTcpHandle(fullServerName, timerExpire, callbackObject, parallel, ref spnBuffer, isIntegratedSecurity);
                     }
+                    // error case for sure
                     else
                     {
-                        SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.ProtocolNotSupportedError, string.Empty);
+                        // when invalid protocol is specified
+                        if (fullServerName.Split(':').Length == 2)
+                        {
+                            if (parallel)
+                            {
+                                SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
+                            }
+                            else
+                            {
+                                SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.ProtocolNotSupportedError, string.Empty);
+                            }
+                        }
+                        // when fullServerName is in invalid format
+                        else
+                        {
+                            SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.INVALID_PROV, 0, SNICommon.InvalidConnStringError, string.Empty);
+                        }
                     }
-                    return null;
+                }
             }
+
+            return sniHandle;
+        }
+
+        private static byte[] MakeMsSqlServerSPN(string fullyQualifiedDomainName, int port = DefaultSqlServerPort)
+        {
+            string serverSpn = SqlServerSpnHeader + "/" + fullyQualifiedDomainName + ":" + port;
+            return Encoding.ASCII.GetBytes(serverSpn);
+        }
+
+        private static string GetFullyQualifiedDomainName(string hostNameOrAddress)
+        {
+            IPHostEntry hostEntry = null;
+            try
+            {
+                hostEntry = Dns.GetHostEntry(hostNameOrAddress);
+            }
+            catch
+            {
+                throw new Exception(SR.reverse_lookup_failed);
```

Comment:
Do we not want to include the original exception here as an inner exception? If native SNI returns a SQL exception in this case, then you can use the ReportSNIError functions here